src/VisualStudio/Core/Def/Implementation/AnalyzerDependency/AnalyzerDependencyCheckingService.cs

```diff
@@ -80,48 +80,57 @@ internal sealed class AnalyzerDependencyCheckingService
 
             var conflicts = results.Conflicts;
             var missingDependencies = results.MissingDependencies;
+            var projects = _workspace.DeferredState.ProjectTracker.ImmutableProjects;
```

Comment:
ah. so before it didn't have race since there can ever be 1 diagnostics per project and UI thread worked as synchronization lock. but with moving things background it now might have since task doesn't guarantee ordering.

might need a queue or something to make sure tasks run in FIFO order.